lib/backend/k8s/resources/kubenetworkpolicy.go

```diff
@@ -91,39 +94,52 @@ func (c *networkPolicyClient) Get(ctx context.Context, key model.Key, revision s
 
 func (c *networkPolicyClient) List(ctx context.Context, list model.ListInterface, revision string) (*model.KVPairList, error) {
 	log.Debug("Received List request on Kubernetes NetworkPolicy type")
-	// List all of the k8s NetworkPolicy objects.
-	networkPolicies := networkingv1.NetworkPolicyList{}
-	req := c.clientSet.NetworkingV1().RESTClient().
-		Get().
-		Resource("networkpolicies")
-	err := req.Do(ctx).Into(&networkPolicies)
-	if err != nil {
-		log.WithError(err).Info("Unable to list K8s Network Policy resources")
-		return nil, K8sErrorToCalico(err, list)
```

Comment:
Does it make any difference going from this code with `RESTClient` to the `NetworkPolicies()` interface?  If not, was it just oversight that this old code was here in this form for so long?

Reply:
Yeah - the code used to use a RESTClient because we wrote this well before client-go had native support for the NetworkPolicy API (e.g., when networkpolicies were in alpha stage).

I think we have just never updated it since.